src/include/math_utils/math_functions.h

```diff
@@ -406,9 +406,33 @@ int64_t binaryPow(int64_t base, int exp);
   @return The powered "number", the same type as `base`
   @sa binaryPow(int64_t base, int exp)
 **/
-template <typename BaseType, typename IntType>
+template <typename BaseType, typename IntType,
+          typename std::enable_if<
+              std::is_constructible<BaseType, int>::value>::type* = nullptr>
+BaseType genericBinaryPow(BaseType base, IntType exp) {
```

Comment:
This function now allows `exp`==0 if `BaseType` can be construct with `BaseType{1}`